tests/checkers/unittest_non_ascii_name.py

```diff
@@ -0,0 +1,312 @@
+            ),
+            pytest.param(
+                """
+                def foobar():
+                    łol="test"  #@
+                """,
+                "Variable",
+                id="function_variable",
+            ),
+            pytest.param(
+                """
+                for łol in os.listdir("."):  #@
+                    ...
+                """,
+                "Variable",
+                id="for_loop_variable",
+            ),
+            pytest.param(
+                """
+                [łoł
+                    for łol in os.listdir(".")  #@
+                ]
+                """,
+                "Variable",
+                id="inline_for_loop_variable",
+            ),
+        ],
+    )
+    def test_assignname(
+        self,
+        code: str,
+        assign_type: str,
+    ):
+        """Variables defined no matter where, should be checked for non ascii"""
+        assign_node = astroid.extract_node(code)
+
+        if not isinstance(assign_node, nodes.AssignName):
+            # For some elements we can't directly extract the assign
+            # node, so we have to manually look in the children for it
+            for child in assign_node.get_children():
+                if isinstance(child, nodes.AssignName):
+                    assign_node = child
+                    break
+
+        # Just to make sure we found the correct node
+        assert isinstance(assign_node, nodes.AssignName)
+
+        with self.assertAddsMessages(
+            pylint.testutils.MessageTest(
+                msg_id="non-ascii-name",
+                node=assign_node,
+                args=(assign_type, "łol"),
+                confidence=pylint.interfaces.HIGH,
+            ),
+            ignore_position=True,
+        ):
+            self.checker.visit_assignname(assign_node)
+
+    def test_class_definition(self):
+        node = astroid.extract_node(
+            """
+            class FooBär: #@
+                ...
+            """
+        )
+        assert isinstance(node, nodes.ClassDef)
+
+        with self.assertAddsMessages(
+            pylint.testutils.MessageTest(
+                msg_id="non-ascii-name",
+                node=node,
+                args=("Class", "FooBär"),
+                confidence=pylint.interfaces.HIGH,
+                line=2,
+                col_offset=0,
+                # pylint: disable-next=fixme
+                # TODO: Maybe we have to select something different
+                #       here, as line 3 is not that thing we expect
+                end_line=3,
+                end_col_offset=7,
+            )
+        ):
+            self.checker.visit_classdef(node)
+
+    def test_import_ignore_star(self):
+        """Special case from xyz import *, where '*' is not A-Za-z0-9 but still valid"""
+        node = astroid.parse("from urllib.parse import *")
+
+        with self.assertAddsMessages():
+            self.walk(node)
+
+    @pytest.mark.parametrize(
```

Comment:
I think it would make sense to convert this into a functional test. Imo, those are a lot better to read and I know they actually test "more" than the unittests. (They almost recreate an instance of `pylint` and run it over the file, whereas the unittests takes some shortcuts to initialise faster. This makes it more prone to missing crucial setup phases).

Reply:
As explained earlier: I still think they are different.
And sorry too much work. Won't do.